src/libs/Navigation/CustomActions.js

```diff
@@ -42,55 +43,83 @@ function navigateBackToRootDrawer(navigationRef) {
 }
 
 /**
- * In order to create the desired browser navigation behavior on web and mobile web we need to replace any
- * type: 'drawer' routes with a type: 'route' so that when pushing to a report screen we never show the
- * drawer. We don't want to remove these since we always want the history length to increase by 1 whenever
- * we are moving to a new report screen or back to a previous one. This is a workaround since
- * react-navigation default behavior for a drawer is to skip pushing history states when navigating to the
- * current route
+ * @param {Object} state
+ * @returns {Object}
+ */
+function getParamsFromState(state) {
+    return lodashGet(state, 'routes[0].state.routes[0].params', {});
```

Comment:
This could be wrong. As the active route is decided via the index property on the state. I think it should be recursive until we get the final Active state.  there could be multiple nested states.

In our case, there are only two nested. 
Stack
  => Drawer || Stack

Reply:
Are you saying it could be wrong when we use the state returned from `getStateFromPath()`? 
Or are you saying the existing logic had some problem and could be wrong?

What should I do to observe this?

Reply:
What I observed from this code is that you are trying to get the params of the active route. 
If you are meant to do that then

So this statement `routes[0].state.routes[0].params` is not necessarily giving us params of active route. Under the state object, any route could be active from all items in the routes array. The state has an index property that tells which one is the active one. 
```state.routes[state.index ?? 0]``` I saw similar code in the `react-navigation` codebase.

Reply:
I think I understand... but there are only two cases where we try to use the state currently:

1. active route which is the just using the root state (logic should be the same as before - was it wrong?)
2. current route which we were previously getting the params and screen name from the custom `navigate()`, but it seemed better to extract that out of `getStateFromPath()` (which I don't think returns any nested state for the report case)

So which case do we need to use recursion for? Is there something I can do in the app to break the current logic and see that what we have isn't working correctly?

Reply:
It seems that it just works for our cases. Probably we don't need to get that deeper into it. I can't think of any cases in our App where this logic won't work if this is extracted from the previous code.

Reply:
Ok gonna leave a note somewhere about this - but maybe not worry about a solution. The custom action will probably need to evolve as time goes on and navigators get more complex.